### PR TITLE
Fix Android compatibility issues

### PR DESCRIPTION
**Pull request by fzhinkin**

### CODE REVIEW

File: build-logic/src/main/kotlin/kotlinx/io/conventions/kotlinx-io-android-compat.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+import ru.vyarus.gradle.plugin.animalsniffer.AnimalSnifferExtension
+
+apply(plugin = "ru.vyarus.animalsniffer")
+
+project.plugins.withType(JavaPlugin::class.java) {
```
Comment (shanshin) [inline build-logic/src/main/kotlin/kotlinx/io/conventions/kotlinx-io-android-compat.gradle.kts:10]: try `plugins.withType(JavaPlugin::class.java).configureEach {`

Comment (shanshin) [reply build-logic/src/main/kotlin/kotlinx/io/conventions/kotlinx-io-android-compat.gradle.kts:10]: Is `plugins.withType` used to configure AnimalSniffer only if the java plugin is specified in the project?

Comment (fzhinkin) [reply build-logic/src/main/kotlin/kotlinx/io/conventions/kotlinx-io-android-compat.gradle.kts:10]: If I correctly understood the question, when the java plugin is not specified, AnimalSniffer plugin's configuration will not affect anything (https://github.com/xvik/gradle-animalsniffer-plugin#compatibility).

Comment (shanshin) [reply build-logic/src/main/kotlin/kotlinx/io/conventions/kotlinx-io-android-compat.gradle.kts:10]: Ok, in this case, the applying of the plugin `apply(plugin = "ru.vyarus.animalsniffer")` can also be moved inside this block.

**Review by shanshin — APPROVED** [review]

**Review by shanshin — APPROVED** [review]

**Pull request merged**